### PR TITLE
Add manual refresh payload to 'spice refresh...'

### PR DESCRIPTION
**Pull request by Jeadie**
## 🗣 Description
 - Add API changes from `` into `spice refresh` CLI command

## 🔨 Related Issues
 - https://github.com/spiceai/gospice/issues/50

## 🔨 Related PRs
 - API changes to spiced: https://github.com/spiceai/spiceai/pull/2515

### CODE REVIEW

File: bin/spice/cmd/refresh.go [hunk 1]
```diff
@@ -17,17 +17,46 @@ limitations under the License.
...
+	refreshSqlFlag  = "refresh-sql"
+	refreshModeFlag = "refresh-mode"
 )
 
 type DatasetRefreshApiResponse struct {
 	Message string `json:"message,omitempty"`
 }
 
+type DatasetRefreshApiRequest struct {
+	SQL  *string `json:"refresh_sql,omitempty"`
```
Comment (phillipleblanc) [inline bin/spice/cmd/refresh.go]:
```suggestion
	RefreshSQL  *string `json:"refresh_sql,omitempty"`
```

Let's be explicit in the code as well, SQL isn't very descriptive.

Comment (phillipleblanc) [reply bin/spice/cmd/refresh.go]: Here and below

File: bin/spice/cmd/refresh.go [hunk 2]
```diff
@@ -17,17 +17,46 @@ limitations under the License.
...
+	Mode *string `json:"refresh_mode,omitempty"`
+}
+
+func construct_request(sql string, mode string) (*string, error) {
```
Comment (phillipleblanc) [inline bin/spice/cmd/refresh.go]:
```suggestion
func constructRequest(sql string, mode string) (*string, error) {
```

Let's stick to the Go style function formatting in Go

File: bin/spice/cmd/refresh.go [hunk 3]
```diff
@@ -40,6 +69,15 @@ spice refresh taxi_trips
 	Run: func(cmd *cobra.Command, args []string) {
 		dataset := args[0]
 
+		sql, _ := cmd.Flags().GetString(refreshSqlFlag)
+		mode, _ := cmd.Flags().GetString(refreshModeFlag)
+
+		// If the mode is not empty, it must be either 'full' or 'append'.
+		if mode != "" && mode != spec.REFRESH_MODE_FULL && mode != spec.REFRESH_MODE_APPEND {
+			cmd.PrintErrln("Invalid refresh mode. Must be 'full' or 'append'")
```
Comment (lukekim) [inline bin/spice/cmd/refresh.go]:
```suggestion
			cmd.PrintErrln("Invalid refresh mode. Valid modes are 'full' or 'append'")
```

File: bin/spice/cmd/refresh.go [hunk 4]
```diff
@@ -17,17 +17,46 @@ limitations under the License.
...
+	Mode *string `json:"refresh_mode,omitempty"`
+}
+
+func constructRequest(sql string, mode string) (*string, error) {
+	r := DatasetRefreshApiRequest{}
+	if sql == "" && mode == "" {
+		return nil, nil
+	}
+	if sql != "" {
+		r.SQL = &sql
```
Comment (lukekim) [inline bin/spice/cmd/refresh.go]:
```suggestion
		r.RefreshSQL = &sql
```

**Review by lukekim — APPROVED** [review]

**Pull request merged**